### PR TITLE
New addon: Rename broadcasts

### PR DESCRIPTION
**Pull request by TheColaber**
Requested in the discord server https://discord.com/channels/806602307750985799/1024221040961929266

image

Currently, you can rename by clicking on the dropdown and choosing "Rename broadcast" which will open a model. This addon is consistent with scratch's method to rename variables. I did not create the option to rename the broadcast by right-clicking on the block to stay consistent.

### CODE REVIEW

**Comment by BroJac5246** [comment]
I feel like pressing "Rename broadcast" is a little tedious. I would personally prefer it if each broadcast had a little pencil icon next to it that would let you rename it. That said, I see why you want consistency.

**Review by apple502j — APPROVED** [review]

**Comment by lisa-wolfgang** [comment]
Is this compatible with the dropdown search addon?

**Comment by apple502j** [comment]
@lisa-wolfgang Think so, see Discord history

**Review by lisa-wolfgang — APPROVED** [review]
[x] Valid reason to merge
[x] To the best of my knowledge, supports all browsers supported by the extension
[x] I personally tested all features in both a Chromium-based browser and Firefox
[x] All added UI elements support both LTR and RTL language modes
[x] I personally reviewed all code line-by-line

[x] All titles, descriptions, info notices, and setting names provided are clear and concise
[x] I personally found the addon to be easy to use during my testing
[x] I personally tested with all other addons and features enabled and did not find compatibility issues
[x] I personally tested with no other addons enabled and did not find issues
[x] The addon is well-integrated with other addons if appropriate

**Comment by lisa-wolfgang** [comment]
One non-merge-blocking issue is that attempting to rename a broadcast to the name of a broadcast that already exists silently fails.

File: addons/rename-broadcasts/addon.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "Rename broadcasts",
```
[the remark this answers is not in the document]
Comment (WorldLanguages) [reply addons/rename-broadcasts/addon.json:2]: Disagree - not using a verb sounds to bad.

File: addons-l10n/en/rename-broadcasts.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "rename-broadcasts/RENAME_BROADCAST": "Rename broadcast",
+  "rename-broadcasts/RENAME_BROADCAST_TITLE": "Rename all \"{name}\" broadcasts to:",
+  "rename-broadcasts/RENAME_BROADCAST_MODAL_TITLE": "Rename Broadcast"
```
Comment (WorldLanguages) [inline addons-l10n/en/rename-broadcasts.json:4]: why u shouting

File: addons/rename-broadcasts/addon.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  "description": "Adds an option to rename broadcasts in the broadcast blocks' dropdown.",
```
Comment (WorldLanguages) [inline addons/rename-broadcasts/addon.json:3]: should be `block's`

Comment (WorldLanguages) [reply addons/rename-broadcasts/addon.json:3]: Or, alternatively, `dropdowns`

File: addons-l10n/en/rename-broadcasts.json (hunk 2 above)
Comment (WorldLanguages) [reply addons-l10n/en/rename-broadcasts.json:4]: nvm

File: addons/rename-broadcasts/addon.json (hunk 3 above)
Comment (apple502j) [reply addons/rename-broadcasts/addon.json:3]:
```suggestion
  "description": "Adds an option to rename broadcasts in the broadcast blocks' dropdowns.",
```

**Comment by WorldLanguages** [comment]
See #5206 for pending TODOs before release.

It would also great to have GarboMuffin review this

**Pull request merged**